src/main/java/com/ryanharter/auto/value/parcel/AutoValueParcelExtension.java

```diff
@@ -83,11 +86,24 @@ String getParcelableCode(Context context, String className, String classToExtend
 
   private void validateProperties(ProcessingEnvironment env,
                                   Map<String, ExecutableElement> properties) {
+    Types typeUtils = env.getTypeUtils();
+    Elements elementUtils = env.getElementUtils();
+    TypeMirror collection = typeUtils.getDeclaredType(
+        elementUtils.getTypeElement(Collection.class.getName()),
```

Comment:
I think you should specifically check for `List` instead of `Collection`. `Parcel` special cases `List` by using the `writeList()` method. Collections on the other hand are only parcelable if their type parameter is serializable.

Check out the implementation of `Parcel#writeValue()`. It has a lot of special cases.